src/Evolution/DiscontinuousGalerkin/MortarData.cpp

```diff
@@ -56,6 +58,134 @@ void MortarData<Dim>::insert_neighbor_mortar_data(
+  ASSERT(local_mortar_data_.has_value(),
+         "Must set local mortar data before setting the local face normal.");
+  const size_t required_storage_size = local_face_normal_magnitude[0].size();
+  if (local_volume_and_face_det_jacobian_and_normal_magnitude_.size() !=
+      required_storage_size) {
+    local_volume_and_face_det_jacobian_and_normal_magnitude_.resize(
+        required_storage_size);
+  }
+  std::copy(local_face_normal_magnitude[0].begin(),
+            local_face_normal_magnitude[0].end(),
+            local_volume_and_face_det_jacobian_and_normal_magnitude_.begin());
+}
+
+template <size_t Dim>
+void MortarData<Dim>::get_local_volume_det_inv_jacobian(
+    const gsl::not_null<Scalar<DataVector>*> local_volume_det_inv_jacobian)
+    const noexcept {
+  ASSERT(local_mortar_data_.has_value(),
+         "Must set local mortar data before getting the local face normal.");
+  ASSERT(
+      local_volume_and_face_det_jacobian_and_normal_magnitude_.size() >
+          2 * std::get<0>(*local_mortar_data()).number_of_grid_points(),
+      "Cannot retrieve the volume inverse Jacobian determinant because it was "
+      "not inserted.");
+  const size_t num_face_points =
+      std::get<0>(*local_mortar_data()).number_of_grid_points();
+  const size_t num_volume_points =
+      local_volume_and_face_det_jacobian_and_normal_magnitude_.size() -
+      2 * num_face_points;
+  get(*local_volume_det_inv_jacobian)
+      .set_data_ref(
+          const_cast<double*>(  // NOLINT
+              local_volume_and_face_det_jacobian_and_normal_magnitude_.data()),
+          num_volume_points);
+}
+
+template <size_t Dim>
+void MortarData<Dim>::get_local_face_det_jacobian(
+    const gsl::not_null<Scalar<DataVector>*> local_face_det_jacobian)
+    const noexcept {
+  ASSERT(local_mortar_data_.has_value(),
+         "Must set local mortar data before getting the local face normal.");
+  ASSERT(
+      local_volume_and_face_det_jacobian_and_normal_magnitude_.size() >
+          2 * std::get<0>(*local_mortar_data()).number_of_grid_points(),
+      "Cannot retrieve the volume inverse Jacobian determinant because it was "
+      "not inserted.");
+  const size_t num_face_points =
+      std::get<0>(*local_mortar_data()).number_of_grid_points();
+  const size_t offset =
+      local_volume_and_face_det_jacobian_and_normal_magnitude_.size() -
+      2 * num_face_points;
+  get(*local_face_det_jacobian)
+      .set_data_ref(
+          // NOLINTNEXTLINE
+          const_cast<double*>(  // NOLINTNEXTLINE
+              local_volume_and_face_det_jacobian_and_normal_magnitude_.data() +
+              offset),
+          num_face_points);
+}
+
+template <size_t Dim>
+void MortarData<Dim>::get_local_face_normal_magnitude(
+    const gsl::not_null<Scalar<DataVector>*> local_face_normal_magnitude)
+    const noexcept {
+  ASSERT(local_mortar_data_.has_value(),
+         "Must set local mortar data before getting the local face normal.");
```

Comment:
normal -> normal magnitude

Reply:
why is there no corresponding assert about non-insertion (as for the other two get functions)?